File: pet_hospital_frontend/src/api/reminder.ts
import api from './index'

export interface Reminder {
    id?: number
    customerId: number
    petId: number
    type: string // VACCINE, BIRTHDAY
    dueDate: string
    status: number // 0:UNSENT, 1:SENT
    sentTime?: string
    customer?: {
        name: string
        phone: string
    }
    pet?: {
        name: string
    }
}

export const reminderApi = {
    list: (params?: { page?: number; size?: number }) =>
        api.get<any, { code: number; data: any }>('/reminders', { params }),

    generate: () =>
        api.post<any, { code: number; data: null }>('/reminders/generate'),

    send: (id: number) =>
        api.post<any, { code: number; data: null }>(`/reminders/send/${id}`)
}

File: pet_hospital_backend/src/main/java/com/pethospital/entity/Reminder.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("reminders")
public class Reminder {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long customerId;
    
    private Long petId;
    
    /**
     * 类型: VACCINE, BIRTHDAY, FOLLOW_UP
     */
    private String type;
    
    private LocalDate dueDate;
    
    /**
     * 状态: 0:未发送, 1:已发送, 2:发送失败
     */
    private Integer status;
    
    private LocalDateTime sentTime;
    
    // 关联字段
    @TableField(exist = false)
    private Customer customer;
    
    @TableField(exist = false)
    private Pet pet;
}
